File: algo_03_two_points/L125.cpp

#include <iostream>
#include <cstring>
using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {
        string ss;
        for(char n:s)
            if(n>='0'&&n<='9')
                ss += n;
            else if(n>='a'&& n<='z')
                ss += n;
            else if(n>='A'&& n<='Z')
                ss += n-'A'+'a';

        int i=0, j=ss.size()-1;
        while(i<j){
            if(ss[i] == ss[j]){
                i++;
                j--;
            }
            else
                return false;
        }
        return true;
    }

};

File: python-algorithm/algo_03_two_points/L125.py
#--coding:utf-8--

class Solution:
    def isPalindrome(self, s: str) -> bool:
        new_s = [c.lower() for c in s if c.isdigit() or c.isalpha()]
        lt, rt = 0, len(new_s)-1
        while lt < rt:
            if new_s[lt] != new_s[rt]:
                return False
            lt += 1
            rt -= 1
        return True